### DISCUSSION TITLE
Transforms are visible even when published on non-standard topics

### DISCUSSION THREAD

**Discussion by aalbaali** [foxglove/community#1007] [General]
I have a bagfile that contains transform messages on the `/tf` and `/tf_static` topics. I usually playback the bags but without playing back all the transforms (e.g., play `odom->base_link` but not `map->odom`). My way to do this is to playback the bag while remapping the transform topics to `/tf:=/bag/tf` and `/tf_static:=/bag/tf_static`, then I have a node that selects the transforms to map from `/bag/tf` to `/tf`.

However, I noticed that when I do that foxglove still views the messages on the remapped topics (i.e., `/bag/tf` and `/bag/tf_static`), which I do *not* want it to do (rviz doesn't do that). It seems that foxgloves publishes the transforms published on *any* topic.

How can I specify which topics foxglove should listen to for the `Transforms` section?

**Comment by defunctzombie** [comment]
This is by design (expected). Foxglove will listen for transforms on all valid topics which publish transforms. There is no way to disable specific transform topics. If you want to prevent Foxglove from consuming a transform topic you'll need to avoid publishing it.

**Reply by aalbaali** [reply]
> all valid topics

What are the valid topics? It seems that foxglove captures any `TFMessage` sent to *any* topic (e.g., `/abcd`).

Thanks for the reply. I find that an odd design choice given that there could be various Transform topics on a bag/node that the user would like to ignore. I guess the decision was made to combine `/tf` and `/tf_static` into a single bundle, but it forces the user into this paradigm.

It seems that my only choice regarding playing back bags is to play the bag, and record the desired transforms into a new bag and then playback the new bag into foxglove. Is there a better way to do this?

**Reply by defunctzombie** [reply]
> What are the valid topics?

Any topic which publishes a TFMessage. "Valid" was not a great word for me to use for this.

Any topic that publishes messages of these schemas: https://docs.foxglove.dev/docs/visualization/panels/3d/#transforms is considered a "transform" topic in the 3D panel and will be subscribed.

Notably the `/tf` convention is a ROS convention rather than a general robotics "thing". Some robots use other topic names for their transforms.

**Comment by deniz-hofmeister** [comment]
Dear @defunctzombie,

thank you for your reply. I was wondering how this can work in a multi robot setup? Right now our design is that we namespace every robot's nodes and topics to its own uuid, but the internal transforms and frame id's are identical, they all have base_link etc. If everything just gets merged you will get a conflict of transforms in a multi robot setup. What can be done here?

Greetings and thank you for your time

**Reply by defunctzombie** [reply]
We don't have support for multi-robot setups like the one you described above. The way the above can be accomplished today is with the frame names to be specific to each robot (when using a single graph approach for your multi-robot setup).

**Comment by jtbandes** [comment]
Internal tracking issue: ERT-302
